File: src/test/samples/validSamples/validSample.hs
-- #region FirstRegion
x = 42
-- #endregion

-- #region Second Region  
data MyClass = MyClass
    --    #region   InnerRegion  
  deriving Show
    --         #endregion    ends InnerRegion  

  --  #region  
deriving UnnamedRegion
--  #endregion  
-- #endregion

File: src/test/samples/validSamples/validSample.m
//#region FirstRegion
#import <Foundation/Foundation.h>
//#endregion

//  #region Second Region
@interface MyClass : NSObject
//  #region    InnerRegion  
- (void)myMethod;
//  #endregion  ends InnerRegion   

//  #region
@property (nonatomic, strong) NSString *name;
// #endregion
@end
// #endregion
